autowarefoundation/autoware.universe
# feat: add vehicle status api

## Description
- add vehicle status api
-  ~~add publish mgrs_grip topic in the map_loader~~ -> move to #3200
- This PR contains 4 api
    - /api/vehicle/kinematics
    - /api/vehicle/status
    ~~ - /api/vehicle/door ~~

## Related links
- https://github.com/autowarefoundation/autoware/issues/3232
- https://github.com/autowarefoundation/autoware_adapi_msgs/pull/24
- https://github.com/autowarefoundation/autoware-documentation/pull/312


## Tests performed
- makesure all those api output the correct data

1. /api/vehicle/kinematic
    - geographic_pose
        - use a map that contain correct MGRS or UTM
        - check whether the latitude and longitude output correctly
        - the altitude should be the same as the pose z value
     - make sure the following have the correct output with the related topic
         - pose = `/localization/kinematic_state` ->pose
         - twist =  `/localization/kinematic_state` ->twist
         - accel = `/localization/acceleration` -> accel

2. /api/vehicle/state
     - make sure the following have the correct output with the related topic
         - steering_tire_angle = `/vehicle/status/steering_status` ->steering_tire_angle
         - gear =  `/vehicle/status/gear_status` ->report
         - turn_indicator = /vehicle/status/turn_indicators_status` -> report
         - hazard_light = /vehicle/status/hazard_lights_status` -> report
         - energy_level = `/vehicle/status/battery_charge` -> energy_level
             - can be test by using following command
             ```bash
             ros2 topic pub /vehicle/status/battery_charge tier4_vehicle_msgs/msg/BatteryStatus energy_level:\ 20
             ```



## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
